### PR TITLE
feat: add dedicated security group module

### PR DESCRIPTION
**Pull request by swibrow**
I paired with @Davidoutz on this to see what would be the simplest implementation of the security group submodule

- created dedicated karpenter security group with some good defaults
- added additional rules to the eks cluster default security group to allow all traffic from the vpc. This is needed for fargate workloads (karpenter) to be accessible from karpenter managed node groups
- made some changes to the cluster access rules to prevent issues when deploying from local. 
- added feature workflow to deploy example/complete if and tf files are changed in the root. This will also be used for renovate now that all terraform prs are combined

### CODE REVIEW

**Review by sgametrio — COMMENTED** [review]
It seems correct to me. Honestly, not an expert, but let's fix forward in case?

File: .github/workflows/terraform-docs.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ name: Generate terraform docs
 on:
   push:
     branches:
-      - renovate**
```
[the remark this answers is not in the document]
Comment (swibrow) [reply .github/workflows/terraform-docs.yaml:7]: it disabled the workflow :)

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -6,6 +6,9 @@ repos:
         args:
           - '--args=-recursive'
       - id: terraform_validate
+        args:
+          - --tf-init-args=-upgrade
+          - --tf-init-args=-lockfile=false
```
[the remark this answers is not in the document]
Comment (swibrow) [reply .pre-commit-config.yaml:11]: it upgrade any of the local terraform when the modules change

File: .github/workflows/terraform-docs.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (sgametrio) [reply .github/workflows/terraform-docs.yaml:7]: @swibrow [minor] Should we rename the file instead to disable it?

File: main.tf [hunk 3]
```diff
@@ -73,7 +70,7 @@ data "aws_iam_roles" "sso" {
 data "aws_iam_roles" "iam_cluster_admins" {
   for_each = var.cluster_admins
 
-  name_regex = "\\b${each.value.role_name}\\b"
+  name_regex = "^${each.value.role_name}$"
```
[the remark this answers is not in the document]
Comment (swibrow) [reply main.tf:73]: it didnt work as expected before. this is now a fairly standard start / finish here regex

File: .pre-commit-config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (sgametrio) [reply .pre-commit-config.yaml:11]: Ah right!! Makes sense

File: .github/workflows/terraform-docs.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (swibrow) [reply .github/workflows/terraform-docs.yaml:7]: I can delete it ;)

**Review by sgametrio — APPROVED** [review]
Approving as it looks good to me :)

**Pull request merged**